test/xpu_api/utilities/pairs/pair.piecewise/piecewise_construct.pass.cpp

```diff
@@ -0,0 +1,102 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include <oneapi/dpl/tuple>
+#include <oneapi/dpl/functional>
+
+#include "support/test_macros.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+class A
+{
+    int i_;
+    char c_;
+
+  public:
+    A(int i, char c) : i_(i), c_(c) {}
+    int
+    get_i() const
+    {
+        return i_;
+    }
+    char
+    get_c() const
+    {
+        return c_;
+    }
+};
+
+class B
+{
+    float f_;
+    unsigned u1_;
+    unsigned u2_;
+
+  public:
+    B(float f, unsigned u1, unsigned u2) : f_(f), u1_(u1), u2_(u2) {}
+    float
+    get_f() const
+    {
+        return f_;
+    }
+    unsigned
+    get_u1() const
+    {
+        return u1_;
+    }
+    unsigned
+    get_u2() const
+    {
+        return u2_;
+    }
+};
```

Comment:
I think `B` is useless. I propose to use something like `dpl::pair<int, A>` for testing.

Reply:
Done